sumpy/fmm.py

```diff
@@ -287,20 +299,33 @@ class SumpyExpansionWrangler:
         Keyword arguments to be passed for handling
         self interactions (source and target particles are the same),
         provided special handling is needed
+
+    .. attribute:: preprocessed_mpole_dtype
+
+        Type for the preprocessed multipole expansion if used for M2L.
     """
 
     def __init__(self, code_container, queue, tree, dtype, fmm_level_to_order,
             source_extra_kwargs,
             kernel_extra_kwargs=None,
             self_extra_kwargs=None,
-            translation_classes_data=None):
+            translation_classes_data=None,
+            preprocessed_mpole_dtype=None):
         self.code = code_container
         self.queue = queue
         self.tree = tree
         self.issued_timing_data_warning = False
 
         self.dtype = dtype
 
+        if not self.code.use_preprocessing_for_m2l:
+            # If not FFT, we don't need complex dtypes
+            self.preprocessed_mpole_dtype = dtype
+        elif preprocessed_mpole_dtype is not None:
+            self.preprocessed_mpole_dtype = preprocessed_mpole_dtype
+        else:
+            self.preprocessed_mpole_dtype = to_complex_dtype(dtype)
```

Comment:
It's weird that the wrangler has to guess this. (If this doesn't want to go easily, add a FIXME comment.)

Reply:
Added a FIXME in 66c1296bd6e701fc0d770b8613cef7bbb6c20a21